### PR TITLE
Reversible tokenizer

### PR DESCRIPTION
**Pull request by mortendahl**
This PR adds support for reversible tokenization based on [AES-SIV](https://tools.ietf.org/html/rfc5297) encryption.

### CODE REVIEW

**Comment by mortendahl** [comment]
@justin1121 @jvmncs would love to get your thinking in on the API

**Comment by justin1121** [comment]
Makes sense to me! Awesome!

File: cape_privacy/pandas/transformations/tokenizer.py [hunk 1]
```diff
@@ -2,6 +2,7 @@
 import secrets
 
 import pandas as pd
+from Crypto.Cipher import AES
```
Comment (jvmncs) [inline cape_privacy/pandas/transformations/tokenizer.py:5]: who made this library and why is it capitalized ☹️

File: cape_privacy/spark/transformations/tokenizer.py [hunk 2]
```diff
@@ -60,3 +60,84 @@ def _to_token(self, x: str):
...
+
+class TokenReverser(base.Transformation):
+    """TokenReverser: recover string from token.
+
+    When applying the TokenReverser to a Spark series of tokens,
+    each token is mapped back to the string that was originally used
+    by ReversibleTokenizer to construct the token. The same key must
+    be used.
+
+    Attributes:
```
Comment (jvmncs) [inline cape_privacy/spark/transformations/tokenizer.py:117]: any reason the `encoding` attribute is missing here?

File: cape_privacy/pandas/transformations/tokenizer.py (hunk 1 above)
Comment (mortendahl) [reply cape_privacy/pandas/transformations/tokenizer.py:5]: yeah, try to look away :)

File: cape_privacy/spark/transformations/tokenizer.py (hunk 2 above)
Comment (mortendahl) [reply cape_privacy/spark/transformations/tokenizer.py:117]: good idea!

**Comment by mortendahl** [comment]
Any thoughts on what's going wrong here? I've updated `requirements.txt` but not sure where/if that causes this problem.

```
ERROR: THESE PACKAGES DO NOT MATCH THE HASHES FROM THE REQUIREMENTS FILE. If you have updated the package versions, please update the hashes. Otherwise, examine the package contents carefully; someone may have tampered with them.
    pyspark[sql]==3.0.0 from https://files.pythonhosted.org/packages/8e/b0/bf9020b56492281b9c9d8aae8f44ff51e1bc91b3ef5a884385cb4e389a40/pyspark-3.0.0.tar.gz#sha256=8c6e5cc51d91eb8d43e81d0b7093292b5e144ac81445491d5f887d2cf4fe121f (from -r requirements.txt (line 16)):
        Expected sha256 8c6e5cc51d91eb8d43e81d0b7093292b5e144ac81445491d5f887d2cf4fe121f
             Got        90a1f98b60966da7f7eaafd70f43f05a77377cc23af30ce53a541e45fb3b77bc

make: *** [bootstrap] Error 1
Makefile:51: recipe for target 'bootstrap' failed
```

**Comment by justin1121** [comment]
Just needed to rerun it must've gotten a bad download... Now you're having codecov problems. Not sure if you want to try adding more tests otherwise we can increase the threshold. I might be unreasonably high at this point.

File: cape_privacy/spark/transformations/tokenizer.py [hunk 3]
```diff
@@ -2,6 +2,7 @@
 import secrets
 
 import pandas as pd
+from Crypto.Cipher import AES
```
[the remark this answers is not in the document]
Comment (mortendahl) [reply cape_privacy/spark/transformations/tokenizer.py:5]: it is (`make fmt`)! not sure what logic isort applies here

**Comment by mortendahl** [comment]
not exactly sure how to get around that failing codecov/patch -- added a few more tests but it's still unhappy. thoughts on merging as-is? @justin1121

**Comment by justin1121** [comment]
Yeah not entirely sure what its trying to say. It looks like the coverage tool can't tell that the `_to_token` and `_from_token` functions are getting tested because they are used in the `series.map`. Lets merge for now and we can consider lowering this number in the future.

**Pull request merged**